### PR TITLE
sqlcipher: new package

### PR DESCRIPTION
**Pull request by mati865**
Closes https://github.com/Alexpux/MINGW-packages/issues/2323

### CODE REVIEW

**Comment by sjemens** [comment]
The sqlite3 is not needed anywhere in compiling. Openssl is also needed in runtime.
I suggest changing to
depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs" "${MINGW_PACKAGE_PREFIX}-readline" "${MINGW_PACKAGE_PREFIX}-openssl")
makedepends=("${MINGW_PACKAGE_PREFIX}-gcc" "tcl")
Also if the wineditline package is installed sqlcipher will try to link to -ledit and the shell.c compilation will fail. So I suggest configure with the following parameters
  ${srcdir}/${_realname}-${pkgver}/configure \
    --prefix=${MINGW_PREFIX} \
    --build=${MINGW_CHOST} \
    --host=${MINGW_CHOST} \
    --enable-tempstore=yes \
    --disable-tcl \
    --disable-editline \
    --enable-readline \
    --with-readline-lib="-L${MINGW_PREFIX}/lib -lreadline" \
    --with-readline-inc="-I${MINGW_PREFIX}/include" \
    CFLAGS="$CFLAGS -DSQLITE_HAS_CODEC" LDFLAGS="-lcrypto"

**Comment by mati865** [comment]
makedepends were taken from Arch, I'll fix it.

> Also if the wineditline package is installed sqlcipher will try to link to -ledit and the shell.c compilation will fail.

I have it installed and it doesn't fail if `-lhistory`  is added to LDFALGS. Can you explain why editline should be disabled?

**Comment by sjemens** [comment]
You are right. With -lhistory (which belongs to package readline) explicity added to LDFLAGS the build does not fail, but i noticed that the sqlcipher.exe does not have the history command capabilities, which after all is why we linked in the first place to the library. If we did not want history capabilities in console, we could disable both editline and readline and skip the dependencies, but as i said, I do not think this is the desired approach.

With -lreadline added to LDFLAGS, everything is working as expected.  With just -ledit added the builing fails (it errors with undefined reference to 'stifle_history'). No need to depend on both since they try to offer the same (or at least I think is the same) functionality.

In conclusion, I think that explicity enabling readline and disabling editline states more clearly what is working and what is not.

**Comment by mati865** [comment]
Updated

**Comment by DX37** [comment]
Thanks, guys! :)
